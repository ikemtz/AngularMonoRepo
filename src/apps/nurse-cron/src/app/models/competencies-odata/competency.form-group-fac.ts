/* eslint-disable @typescript-eslint/no-unused-vars */
/* istanbul ignore file */
/**
 * This file is generated by the openapi-ts-generator
 * #form-group-factory.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
import { FormControl, FormArray, FormGroup, Validators } from '@angular/forms'; //NOSONAR
import { ICompetencyForm } from './competency.form';

export function CompetencyFormGroupFac(): FormGroup<ICompetencyForm> {
  return new FormGroup<ICompetencyForm>({
    id: new FormControl<string | null | undefined>(null),
    name: new FormControl<string | null | undefined>(null),
    isEnabled: new FormControl<boolean | null | undefined>(null),
  });
}
