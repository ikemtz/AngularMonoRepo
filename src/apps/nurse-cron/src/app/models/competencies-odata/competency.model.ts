/* istanbul ignore file */
/**
 * This file is generated by the openapi-ts-generator
 * #model.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface ICompetency {
  id?: string | null;
  name?: string | null;
  isEnabled?: boolean | null;
}
