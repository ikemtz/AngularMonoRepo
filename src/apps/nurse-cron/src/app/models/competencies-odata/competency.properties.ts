/* istanbul ignore file */
/**
 * This file is generated by the openapi-ts-generator
 * #model-properties.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
export enum CompetencyProperties {
  ID = 'id',
  NAME = 'name',
  IS_ENABLED = 'isEnabled',
}
