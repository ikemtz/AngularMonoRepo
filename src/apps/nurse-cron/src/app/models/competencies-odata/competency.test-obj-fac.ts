/* istanbul ignore file */
/**
 * This file is generated by the openapi-ts-generator
 * #test-object-factory.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */

import { CompetencyProperties } from './competency.properties';

export function createTestCompetency() {
    return { 
      [CompetencyProperties.ID]: 'ID',
      [CompetencyProperties.NAME]: 'NAME',
      [CompetencyProperties.IS_ENABLED]: false, 
    };
}
