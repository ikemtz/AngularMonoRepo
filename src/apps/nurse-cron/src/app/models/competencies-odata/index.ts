/* istanbul ignore file */
/**
 * This file is generated by the openapi-ts-generator
 * #generate-barrel-ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
export * from './competency.form-group-fac';
export * from './competency.form';
export * from './competency.model';
export * from './competency.properties';
export * from './competency.test-obj-fac';
