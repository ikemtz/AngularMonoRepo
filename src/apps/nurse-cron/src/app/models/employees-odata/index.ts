/* istanbul ignore file */
/**
 * This file is generated by the openapi-ts-generator
 * #generate-barrel-ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
export * from './employee-certification.form-group-fac';
export * from './employee-certification.form';
export * from './employee-certification.model';
export * from './employee-certification.properties';
export * from './employee-certification.test-obj-fac';
export * from './employee-competency.form-group-fac';
export * from './employee-competency.form';
export * from './employee-competency.model';
export * from './employee-competency.properties';
export * from './employee-competency.test-obj-fac';
export * from './employee-health-item.form-group-fac';
export * from './employee-health-item.form';
export * from './employee-health-item.model';
export * from './employee-health-item.properties';
export * from './employee-health-item.test-obj-fac';
export * from './employee.form-group-fac';
export * from './employee.form';
export * from './employee.model';
export * from './employee.properties';
export * from './employee.test-obj-fac';
