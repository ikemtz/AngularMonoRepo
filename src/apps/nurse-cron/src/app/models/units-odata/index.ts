/* istanbul ignore file */
/**
 * This file is generated by the openapi-ts-generator
 * #generate-barrel-ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
export * from './building.form-group-fac';
export * from './building.form';
export * from './building.model';
export * from './building.properties';
export * from './building.test-obj-fac';
export * from './unit.form-group-fac';
export * from './unit.form';
export * from './unit.model';
export * from './unit.properties';
export * from './unit.test-obj-fac';
