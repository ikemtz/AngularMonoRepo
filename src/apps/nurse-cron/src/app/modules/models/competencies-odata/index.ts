/**
 * This file is generated by the openapi-ts-generator
 * #generate-barrel-ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
export * from './competency.form-group-fac';
export * from './competency.model';
export * from './competency.properties';
