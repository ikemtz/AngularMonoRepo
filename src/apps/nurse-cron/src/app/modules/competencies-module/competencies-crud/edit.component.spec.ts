import { CUSTOM_ELEMENTS_SCHEMA, NO_ERRORS_SCHEMA } from '@angular/core';
import { waitForAsync, ComponentFixture, TestBed } from '@angular/core/testing';
import { ReactiveFormsModule } from '@angular/forms';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';
import { createDataEntryMockFacade } from 'imng-kendo-data-entry/testing';
import { of } from 'rxjs';
import { mockConsoleError } from 'imng-ngrx-utils/testing';

import { CompetencyEditComponent } from './edit.component';
import { CompetencyCrudFacade } from './crud.facade';
import { CompetencyProperties, ICompetency } from '../../../models/competencies-odata';

describe('CompetencyEditComponent', () => {
  let component: CompetencyEditComponent;
  let fixture: ComponentFixture<CompetencyEditComponent>;
  let facade: CompetencyCrudFacade;

  beforeEach(waitForAsync(() => {
    TestBed.configureTestingModule({
      declarations: [CompetencyEditComponent],
      imports: [ReactiveFormsModule, NoopAnimationsModule,],
      providers: [{ provide: CompetencyCrudFacade, useValue: createDataEntryMockFacade({ currentEntity$: of({}) }) }],
      schemas: [CUSTOM_ELEMENTS_SCHEMA, NO_ERRORS_SCHEMA],
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(CompetencyEditComponent);
    component = fixture.componentInstance;
    facade = TestBed.inject(CompetencyCrudFacade);
    fixture.detectChanges();
  });

  afterAll(() => {
    component.ngOnDestroy();
  });

  test('should update', () => {
    component.initForm();
    component.addEditForm.patchValue({
      [CompetencyProperties.ID]: 'ID',
      [CompetencyProperties.NAME]: 'NAME',
      [CompetencyProperties.IS_ENABLED]: true,
    });
    let item: ICompetency | undefined;
    facade.updateExistingEntity = jest.fn(x => (item = x));
    expect(component.getFormErrors()).toStrictEqual([]);
    component.onSubmit();
    expect(facade.saveNewEntity).toBeCalledTimes(0);
    expect(facade.updateExistingEntity).toBeCalledTimes(1);

    expect(item).toMatchSnapshot();

  });

  /**
   * Note: if this test fails, then you're missing validators in your forms.
   * Using form validators is typically a good idea.
   */
  test('should not update', () => {
    const consoleErrorMock = mockConsoleError();
    component.addEditForm?.patchValue({});
    component.onSubmit();
    expect(facade.saveNewEntity).toBeCalledTimes(0);
    expect(facade.updateExistingEntity).toBeCalledTimes(1);
    consoleErrorMock.mockRestore();
  });

  test('should cancel', () => {
    component.cancel();
    expect(facade.clearCurrentEntity).toBeCalledTimes(1);
  });
});
