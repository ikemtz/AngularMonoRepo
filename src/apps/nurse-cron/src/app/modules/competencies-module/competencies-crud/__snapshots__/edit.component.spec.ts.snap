// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompetencyEditComponent should update 1`] = `
{
  "id": "ID",
  "isEnabled": true,
  "name": "NAME",
}
`;
